### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by himadri2110**
Authentication for the e-commerce app

Pages:
- Login 
- SignUp
- Logout

Toggle password visibility
Auth context 
Conditional Rendering on Navbar

Test credentials for Login:
email: adarshbalika@gmail.com
password: adarshbalika

### CODE REVIEW

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import { Navbar } from "../../../components/Navbar/Navbar";
+import { Footer } from "../../../components/Footer/Footer";
```
Comment (manojasarsa) [inline src/pages/Authentication/SignUp/SignUp.jsx:2]: You can make use of barrel export, it will be more readable and easy to refactor(if needed).

**Review by manojasarsa — APPROVED** [review]
The code is readable and looks clean.
Good to go!

File: src/pages/Authentication/SignUp/SignUp.jsx (hunk 1 above)
Comment (himadri2110) [reply src/pages/Authentication/SignUp/SignUp.jsx:2]: Yes, I will implement this in a separate branch. Thank you

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=himadri
```
Comment (adidoshi) [inline .env]: You can add the .env file to your .gitignore file so it will not get uploaded on github.

File: src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -13,10 +16,22 @@ const Navbar = () => {
...
+            <i className="fa-solid fa-sign-in"></i>
+            <span>Login</span>
+          </Link>
+        ) : (
+          <Link
+            to="/logout"
+            className="icon logout"
+            title="Logout"
+            onClick={() => setIsAuth(false)}
+          >
```
Comment (adidoshi) [inline src/components/Navbar/Navbar.jsx]: As you are setting - token to localStorage while login, then when user logs out you can remove it from localStorage.

File: src/pages/Authentication/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+
+    try {
+      const { data } = await loginService(loginInput);
+
+      localStorage.setItem("token", data.encodedToken);
+      setToken(data.encodedToken);
+
```
Comment (adidoshi) [inline src/pages/Authentication/Login/Login.jsx:28]: You can redirect the user to a different page, when the user logs in.
```js
// Just an example, you can try - 
 const navigate = useNavigate()
onClick={() => navigate('your-route')}
```

File: src/Routes/NavRoutes.jsx [hunk 5]
```diff
@@ -13,6 +16,9 @@ const NavRoutes = () => {
       <Route path="/products" element={<Products />} />
       <Route path="/cart" element={<Cart />} />
       <Route path="/wishlist" element={<Wishlist />} />
```
Comment (adidoshi) [inline src/Routes/NavRoutes.jsx:18]: Maybe you can use private route here for user specific routes: 
```js
<Route
            path="your-route"
            element={
              <RequireAuth>
                <YourComponent/>
              </RequireAuth>
            }
          />
```

**Review by adidoshi — APPROVED** [review]
Rest the code looks clean!

File: src/components/Navbar/Navbar.jsx (hunk 3 above)
Comment (himadri2110) [reply src/components/Navbar/Navbar.jsx]: Yes, I forgot to delete the token :') Thank you

File: src/pages/Authentication/Login/Login.jsx (hunk 4 above)
Comment (himadri2110) [reply src/pages/Authentication/Login/Login.jsx:28]: Yes, I will implement this. Thanks for elaborating on the example.

File: src/Routes/NavRoutes.jsx (hunk 5 above)
Comment (himadri2110) [reply src/Routes/NavRoutes.jsx:18]: Yes, I will implement this while making Cart and Wishlist, in their respective PRs. Thank you.

File: .env (hunk 2 above)
Comment (himadri2110) [reply .env]: Yes, thank you for pointing it out.

**Pull request merged**